File: dropwizard-scala_2.9.1/src/test/scala/com/yammer/dropwizard/examples/HelloWorldResource.scala
package com.yammer.dropwizard.examples

import javax.ws.rs._
import core.Response.Status
import core.{Response, MediaType}
import com.yammer.metrics.annotation.Timed

@Path("/hello-world")
@Produces(Array(MediaType.APPLICATION_JSON))
class HelloWorldResource(saying: String) {
  @GET
  @Timed
  def sayHello(@QueryParam("opt") opt: Option[String]) = Seq(saying)

  @POST
  @Timed
  def intentionalError = Response.status(Status.BAD_REQUEST).build()

  @PUT
  @Timed
  def unintentionalError = None.get
}

File: dropwizard-views/src/test/java/com/yammer/dropwizard/views/example/HelloWorldResource.java
package com.yammer.dropwizard.views.example;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

@Path("/hello")
@Produces(MediaType.TEXT_HTML)
public class HelloWorldResource {
    @GET
    public HelloWorldView show(@QueryParam("name") @DefaultValue("Stranger") String name) {
        return new HelloWorldView(new Person(name));
    }
}
